### PR TITLE
Tickets/dm 16501

### PR DESCRIPTION
**Pull request by natelust**

### CODE REVIEW

**Comment by laurenam** [comment]
Looks like this is now behind master (**Remove the unused MultiBandDataIdContainer** has already been merged: https://github.com/lsst/pipe_drivers/commit/9411d8c0da6ab7365ecea8980bd9f513ffff7577 (just in case you hadn't noticed).

File: python/lsst/pipe/drivers/multiBandDriver.py [hunk 1]
```diff
@@ -192,6 +150,10 @@ def __init__(self, butler=None, schema=None, refObjLoader=None, reuse=tuple(), *
             self.measureCoaddSources.schema))
         self.makeSubtask("forcedPhotCoadd", refSchema=afwTable.Schema(
             self.mergeCoaddMeasurements.schema))
+        if self.config.fakesAdded:
+            self.coaddType = "fakes_" + self.config.coaddName
```
Comment (natelust) [inline python/lsst/pipe/drivers/multiBandDriver.py:150]: Consider string formatting

File: python/lsst/pipe/drivers/skyCorrection.py [hunk 2]
```diff
@@ -45,6 +45,7 @@ class SkyCorrectionConfig(Config):
     doBgModel = Field(dtype=bool, default=True, doc="Do background model subtraction?")
     doSky = Field(dtype=bool, default=True, doc="Do sky frame subtraction?")
     binning = Field(dtype=int, default=8, doc="Binning factor for constructing focal-plane images")
+    fakesAdded = Field(dtype=bool, default=False, doc="Have fakes been added?")
```
Comment (natelust) [inline python/lsst/pipe/drivers/skyCorrection.py]: See comment in another repo about the name fakes added, I think a different name like fakesPresent or something like that is closer to the meaning you had in mind, but I dont have very strong feeling on this

File: python/lsst/pipe/drivers/skyCorrection.py [hunk 3]
```diff
@@ -113,14 +114,19 @@ def runDataRef(self, expRef):
         if DEBUG:
             extension = "-%(visit)d.fits" % expRef.dataId
 
+        if self.config.fakesAdded:
```
Comment (natelust) [inline python/lsst/pipe/drivers/skyCorrection.py]: This code block should not be duplicated twice, consider moving it to init and assigning to self, either with the name calexpType or something else if you are worried about that name being used elsewhere

File: python/lsst/pipe/drivers/coaddDriver.py [hunk 4]
```diff
@@ -157,10 +163,14 @@ def runDataRef(self, tractPatchRefList, butler, selectIdList=[]):
         self.log.info("Non-empty tracts (%d): %s" % (len(tractPatchRefList),
                                                      [patchRefList[0].dataId["tract"] for patchRefList in
                                                       tractPatchRefList]))
+        if self.config.fakesAdded:
```
Comment (natelust) [inline python/lsst/pipe/drivers/coaddDriver.py]: Dont have this twice, consider making it class level

File: python/lsst/pipe/drivers/coaddDriver.py [hunk 5]
```diff
@@ -35,13 +35,19 @@ class CoaddDriverConfig(Config):
...
         target=DetectCoaddSourcesTask, doc="Detect sources on coadd")
+    fakesAdded = Field(dtype=bool, default=False, doc="Have fakes been added?")
 
     def setDefaults(self):
         self.makeCoaddTempExp.select.retarget(NullSelectImagesTask)
         self.assembleCoadd.select.retarget(NullSelectImagesTask)
         self.assembleCoadd.doWrite = False
 
     def validate(self):
+        if self.fakesAdded:
```
Comment (natelust) [inline python/lsst/pipe/drivers/coaddDriver.py]: I don't think validate should be used to assign anything itself, it should only be used to check if options were set in a consistent way.

File: python/lsst/pipe/drivers/coaddDriver.py [hunk 6]
```diff
@@ -35,13 +35,19 @@ class CoaddDriverConfig(Config):
                         doc="Run detection on the coaddition product")
     detectCoaddSources = ConfigurableField(
         target=DetectCoaddSourcesTask, doc="Detect sources on coadd")
+    fakesAdded = Field(dtype=bool, default=False, doc="Have fakes been added?")
```
Comment (natelust) [inline python/lsst/pipe/drivers/coaddDriver.py]: This needs a better doc, and consider a different name, see comments on related ticket

**Review by natelust — COMMENTED** [review]
I cant seem to hit the request changes, but I would like to see a few before this is approved

**Pull request merged**